podman_compose.py

```diff
@@ -1528,7 +1528,7 @@ def normalize_service_final(service: dict, project_dir: str) -> dict:
         context = build if is_str(build) else build.get("context", ".")
         context = os.path.normpath(os.path.join(project_dir, context))
         dockerfile = (
-            "Dockerfile" if is_str(build) else service["build"].get("dockerfile", "Dockerfile")
+            None if is_str(build) else service["build"].get("dockerfile", None)
         )
         if not is_dict(service["build"]):
             service["build"] = {}
```

Comment:
This works, but aren't we doing unnecessary work? It seems we would get to the same place if we just removed the `dockerfile` logic completely, like this:

```
def normalize_service_final(service: dict, project_dir: str) -> dict:
    if "build" in service:
        build = service["build"]
        context = build if is_str(build) else build.get("context", ".")
        context = os.path.normpath(os.path.join(project_dir, context))
        if not is_dict(service["build"]):
            service["build"] = {}
        service["build"]["context"] = context
    return service
```

This will not add a `dockerfile` key if one did not already exist, and that will allow the logic in `build_one` to find the correct file.

Reply:
This would prevent setting a custom dockerfile like this:
`build:
  context: .
  dockerfile: webapp.Dockerfile`

[spec](https://docs.docker.com/compose/compose-file/build/)

Reply:
No it doesn't, because of:

```
build = service["build"]
```

If `service['build']` was a dictionary at the start of the function, it will still be one at the end of the function.

Reply:
Yes you are right, I read the code a bit too fast. I'm removing it now then.

Reply:
The tests are expecting a dockerfile to be present in every case, they should also be updated to reflect this change.